developer/src/kmdecomp/savekeyboard.cpp

```diff
@@ -369,17 +369,18 @@ void wr(FILE *fp, PWSTR buf)
 	fwrite(buf, wcslen(buf) * 2, 1, fp);
 }
 
-int SaveKeyboardSource(LPKEYBOARD kbd, LPBYTE lpBitmap, DWORD cbBitmap, char *filename, char *bmpfile)
+int SaveKeyboardSource(LPKEYBOARD kbd, LPBYTE lpBitmap, DWORD cbBitmap, char* filename, char* bmpfile)
 {
-	PWCHAR buf;
-	FILE *fp;
-	LPSTORE sp;
-	LPGROUP gp;
-	LPKEY kp;
-	unsigned int i, j;
-  char bmpbuf[256];
+  PWCHAR buf;
+  FILE* fp;
+  LPSTORE sp;
+  LPGROUP gp;
+  LPKEY kp;
+  unsigned int i, j;
+  char bmpbuf[_MAX_PATH];
+  char bmp_drive[_MAX_DRIVE], bmp_dir[_MAX_DIR], bmp_filename[_MAX_FNAME], bmp_ext[_MAX_EXT];
 
-	buf = new WCHAR[1024];
+	buf = new WCHAR[2048];
```

Comment:
Is there a reason we can't do this?

```suggestion
  buf = new WCHAR[BUFSIZE];
```

Then again, I see the original version was actually twice the original value for `BUFSIZE`, so maybe it should have a separate `#define`?

Also, apologies for the indentation 'change' - it seems raw spaces don't match whatever was there before.

Reply:
Whoa, heavy change from how it looked on the files tab.  Definitely must be tab characters.

Reply:
This code probably has tabs not spaces, being quite old. Don't stress over it 😁

Reply:
In reality, it shouldn't be a static buffer but rather allocated based on the file content. But that's a much bigger job and I don't want to get into that at present!